source/extensions/priv/server/ntds_jet.c

```diff
@@ -0,0 +1,363 @@
+	}
+	//Convert the FILETIME to a SYSTEMTIME so we can get a human readable date
+	FileTimeToSystemTime(&accountExpiry, &accountExpiry2);
+	int dateResult = GetDateFormat(LOCALE_SYSTEM_DEFAULT, DATE_LONGDATE, &accountExpiry2, NULL, userAccount->expiryDate, 255);
+	// Getting Human Readable will fail if account never expires. Just set the expiryDate string to 'never'
+	if (dateResult == 0){
+		strncpy(userAccount->expiryDate, "Never", 6);
+	}
+	// Grab the last logon date and time
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->lastLogon.columnid, &lastLogon, sizeof(lastLogon), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	//Convert the FILETIME to a SYSTEMTIME so we can get a human readable date
+	FileTimeToSystemTime(&lastLogon, &lastLogon2);
+	dateResult = GetDateFormat(LOCALE_SYSTEM_DEFAULT, DATE_LONGDATE, &lastLogon2, NULL, userAccount->logonDate, 255);
+	// Getting Human Readable will fail if account has never logged in, much like the expiry date
+	if (dateResult == 0){
+		strncpy(userAccount->logonDate, "Never", 6);
+	}
+	dateResult = GetTimeFormat(LOCALE_SYSTEM_DEFAULT, 0, &lastLogon2, NULL, userAccount->logonTime, 255);
+	if (dateResult == 0){
+		strncpy(userAccount->logonTime, "Never", 6);
+	}
+	// Grab the last password change date and time
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->lastPasswordChange.columnid, &lastPass, sizeof(lastPass), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	//Convert the FILETIME to a SYSTEMTIME so we can get a human readable date
+	FileTimeToSystemTime(&lastPass, &lastPass2);
+	dateResult = GetDateFormat(LOCALE_SYSTEM_DEFAULT, DATE_LONGDATE, &lastPass2, NULL, userAccount->passChangeDate, 255);
+	// Getting Human Readable will fail if account has never logged in, much like the expiry date
+	if (dateResult == 0){
+		strncpy(userAccount->passChangeDate, "Never", 6);
+	}
+	dateResult = GetTimeFormat(LOCALE_SYSTEM_DEFAULT, 0, &lastPass2, NULL, userAccount->passChangeTime, 255);
+	if (dateResult == 0){
+		strncpy(userAccount->passChangeTime, "Never", 6);
+	}
+	// Grab the Account Description here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->accountDescription.columnid, &userAccount->accountDescription, sizeof(userAccount->accountDescription), &columnSize, 0, NULL);
+	if (readStatus == JET_wrnColumnNull){
+		memset(userAccount->accountDescription, 0, sizeof(userAccount->accountDescription));
+	}
+	else if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	// Grab the UserAccountControl flags here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->accountControl.columnid, &accountControl, sizeof(accountControl), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	if (accountControl & NTDS_ACCOUNT_DISABLED){

```

Comment:
Not a big deal, but a trick for converting flags into bools is to either:
1. use stdbool.h - bools auto-coerce into 1 or 0 magically! Problem is ABI doesn't define sizeof(bool), so its not safe to transmit over the wire in a platform-independent way, though this is arguably not a problem since this is Windows-specific.
2. userAccount->noPassword = !!(accountControl & NTDS_ACCOUNT_DISABLED);